### PR TITLE
feat(training,#14579): OOT protocol instrument -- precision DM jambe + train-end freeze

### PR DESCRIPTION
**Pull request by jsboige**
## feat(training,#14579): OOT protocol instrument — precision DM jambe + train-end freeze

Grain: MED/training -- lane myia-po-2027:CoursIA -- prev: LIGHT/ledger #14563

### Périmètre

Issue #14579 (sous-grain de l'umbrella #1454) : produire le premier verdict
**out-of-time réel** pour L4 Decision Transformer — entraînement gelé à une borne,
holdout postérieur absent de tous les folds et de toute sélection d'hyperparamètres.

Cette PR livre **l'instrument** du protocole (§C renforcé #11010). Le verdict
lui-même (run GPU label `oot` en cours) est un grain séparé qui mettra à jour
`REGISTRY.md` et la section verdict de la doc.

### Deltas

| Fichier | Changement |
|---|---|
| `scripts/validate_xrp_dt_holdout.py` | **Jambe de précision §C** : erreur directionnelle `e_t = position_t − sign(r_{t+1})` — DM `mse`/`mae` sign-aware pour des positions ±1 (le mse direct sur les retours positionnés est sign-blind, `(−r)² = r²`, cf #10228). DM `linear` conservé avec `role: bias_control` explicite. Biais signés par modèle (`mean(return)` DT / momentum / BH) rapportés par seed. Verdict = conjonction `edge ≥ 2σ` cross-seed **ET** `dm_p_mse_median < 0.05` (mae en variante), sinon `NO-BEATS`/`INCONCLUSIVE`. |
| `scripts/train_dt_multiseed.py` | `--train-end <date>` : tronque le CSV à la borne gelée inclusive avant le walk-forward, `data_hash` recalculé sur la tranche gelée. |
| `docs/L4_decision_transformer.md` | Section « Protocole OOT (#14579) » : brique par brique (gel, jambe, contrôle, conjonction) + dataset de référence. |

### Validation

| Étape | Résultat |
|---|---|
| Smoke CPU holdout (`--smoke`) | PASS — les 3 jambes DM rendent (mse 0.78, mae 0.69, linear-ctrl 0.066 sur données synthétiques, 2 seeds × 2 epochs), biais signés rapportés, verdict NO-BEATS cohérent |
| Dry-run WF (`--dry-run --train-end 2025-06-30`) | `--train-end` parse + troncature + walk-forward OK (5/10 coins exécutés, puis abandon volontaire du smoke) |
| Diff doc vs main | 17 insertions pures, 0 suppression (aucun churn d'endings) |
| Pre-commit hooks | 9/9 PASS (gitleaks, H.3 n/a scripts, #13326) |

### Dataset de référence (gitignored, prototype #14579)

`datasets/yfinance/crypto_panier/XRP-USD.csv` — bornes `2018-01-01 → 2026-09-04`
(3169 rows), sha256 `71d8aee9d0bda18b`, téléchargé yfinance 04/09.

### Run en cours

`validate_xrp_dt_holdout.py --label oot --train-end 2025-06-30 --seeds 0 1 7 42 99
--epochs 30` sur GPU (RTX 4060). Le verdict (et les chiffres par seed) sera rapporté
dans la PR suivante : mise à jour de la section OOT de la doc + `REGISTRY.md` ligne L4.

### Notes

- `loss_fn="linear"` reste la jambe « contrôle de biais » uniquement, jamais la
  conjonction (§C amendé #11010) — la doc `dm_test.py` l'explicite déjà.
- Le churn lié au CRLF de la doc a été éliminé avant push (diff vérifié
  whitespace-insensible = 0 suppression).

See #14579

### CODE REVIEW

**Review by clusterManager-Myia — COMMENTED** [review]
**[NanoClaw]** — structural review : OOT protocol instrument (jambe de précision §C + gel de borne), 3 fichiers +107/−14

**Favorable — vérifié au code, pas au body.** Au head `d5b7ea5a76` :

- **Jambe de précision §C** (`validate_xrp_dt_holdout.py` l.153-183) : construction `e_t = position_t − sign(r_t)` sign-aware correcte — politique alignée → 0, opposée → ±2, flat vs mouvement → ±1 ; l'alignement `target_sign[i] = sign(test_returns_full[i])` est cohérent avec la convention `net_returns` (position i tenue pendant le rendement i), donc l'erreur mesure bien la précision directionnelle de chaque politique. DM mse/mae calculés DT vs BH **et** DT vs momentum, `linear` conservé avec `role: bias_control` explicite — exactement la séparation revendiquée (jambe de conjonction ≠ contrôle de biais, #11010).
- **Conjonction verdict** (l.325-331) : `NO-BEATS` si moyenne ≤ BH, sinon `BEATS` exige seeds_beat ≥ min(4,n) **ET** edge_sigma ≥ 2 **ET** dm_p_mse_median < 0.05 — hiérarchie saine (pas de « promising »), médiane cross-seed pour la jambe DM.
- **Gel de borne** (`train_dt_multiseed.py` l.105-113) : troncature `raw.index <= train_end` **avant** le walk-forward, `data_hash` recalculé sur la tranche gelée — le holdout est structurellement invisible des folds. ✓ claim du body.
- **Dépendance** : `dm_test.py` présent au head (7 287 bytes) — l'import n'est pas fantôme.
- **Recomptage croisé avec #14630 (le verdict)** : `edge_sigma = (−1.744 − (−0.592)) / 0.214 = −5.38` — **exactement** la valeur du tableau OOT de #14630, et le verdict NO-BEATS suit la branche `mean ≤ BH` (−1.744 ≤ −0.592). La chaîne instrument → verdict est cohérente de bout en bout.
- Pas de secret dans les diffs (gitleaks 9/9, confirmé par scan local).

**2 points mineurs, non bloquants :**

1. **Le code applique 3 conjoncts, le body/doc en citent 2** : `seeds_beat ≥ min(4, len)` participe au verdict (l.326-328) mais le body dit « edge ≥ 2σ ET dm_p_mse_median < 0.05 ». Le header du script, lui, documente les trois (« ≥4 seeds, edge ≥ 2σ sinon INCONCLUSIVE »). Sur #14630 les trois échouent de toute façon (0/5 seeds), mais la section « Protocole OOT » de la doc gagnerait à citer le 3ᵉ conjonct — c'est celui qui protège contre 1 seed chanceux sur 5.
2. **Cosmétique** : l.111-113 de `train_dt_multiseed.py`, le `if/else` calcule `data_hash` dans les deux branches — seule la troncature mérite la branche. Zéro impact.

Lecture d'ensemble : ce couple #14629/#14630 est le genre de livraison qu'on veut voir — résultat négatif assumé (3ᵉ non-reproduction temporelle), instrument re-exécutable, séries par seed persistées pour recalcul DM post hoc (le commentaire l.189-192 montre que la leçon du p-value gelé au buggy mse a été intégrée). #14630 (le verdict) roule au prochain slot pour sa review propre.

Review structurelle (fenêtre contexte) : 2 fichiers porteurs lus en entier sur 3 changés ; le 3ᵉ (`docs/L4_decision_transformer.md`, +17) non lu ligne à ligne.
